### ISSUE TITLE
Support for document streams, objects, objects in subarrays

### ISSUE THREAD

**Issue by OnkelTem** [MAXakaWIZARD/JsonCollectionParser#3]
Currently the `\JsonCollectionParser\Listener` supports only:

* objects inside array: `[ { } , { } , { } , ... ]`

It seems reasonable to add support for more forms of input data, at least object-oriented:

* object: `{ }`
* stream of objects: `{ } { } { } ...`
* objects in subarrays: `[ [ { } , { } ,   ... ] ]`
* and combination of above, e.g: `[ { } , { } ] { }`

Except for the first case, the data is actually a concatenation of json documents, a frequent case when working with stream data.

The underlying library unfortunately didn't support for multiple documents, hence this PR:
https://github.com/salsify/jsonstreamingparser/issues/60

After and if it's accepted, please check out and evaluate this branch: https://github.com/OnkelTem/JsonCollectionParser/tree/documents-stream-support
It implements the mentioned cases.

**Comment by OnkelTem** [comment]
Ok, I've changed my mind and created a PR (https://github.com/MAXakaWIZARD/JsonCollectionParser/pull/4): please do not accept it until the upstream is updated as described. The tests WILL fail until then.

**Comment by MAXakaWIZARD** [comment]
This stream of objects is not valid JSON `{ } { } { } ...`, so let's not implement it

**Comment by MAXakaWIZARD** [comment]
This one is also not valid JSON: `[ { } , { } ] { }`

**Comment by OnkelTem** [comment]
@MAXakaWIZARD 

> This stream of objects is not valid JSON { } { } { } ..., so let's not implement it
> This one is also not valid JSON: [ { } , { } ] { }

Strictly speaking json which you would probably consider valid, i.e.
```
[{"a": 1}, {"b": 1}
```
is not valid either.  Basically, when you have a list of objects (say, in a database) there is no clear way to join them in a "valid" json. So it won't be valid in either case. The whole idea behind stream parsers is to work on "unfinished" data. 

Second, the mentioned sequence `{ } { } { }` is supposed to be THREE valid jsons, while the second case has TWO valid jsons and that I marked specifically in: *data is actually a concatenation of json documents, a frequent case when working with stream data.*

In other words, a stream parser should support a sequence of valid objects otherwise we would have to invent one more chain in line to pre-parse the stream to make the parser feel happy - strange idea by itself, isn't it?

**Comment by OnkelTem** [comment]
@MAXakaWIZARD https://github.com/salsify/jsonstreamingparser/issues/62

**Comment by MAXakaWIZARD** [comment]
@OnkelTem thanks for your help, v1.6.0 has been released: https://github.com/MAXakaWIZARD/JsonCollectionParser/blob/master/CHANGELOG.md
